### PR TITLE
DO-17 :whale: :sparkles: Add Dockerfiles, dockerignore and docs

### PR DESCRIPTION
**Pull request by dkravetz**
### References

[DO-17](https://xainag.atlassian.net/browse/DO-17) [XP-305
](https://xainag.atlassian.net/browse/XP-305)
### Summary

Create Dockerfiles for xain-fl

### Are there any open tasks/blockers for the ticket?

No

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +77,39 @@ For more information about the CLI and its arguments, run:
...
+Development image
+---
+
+To run the coordinator's development image, first build the Docker image:
+
+```shell
+$ docker build -t xain-fl-dev -f dev.dockerfile .
+```
+
+Then run the image, mounting the directory as a Docker volume, and call the
```
Comment (bartlomiej-matejczy) [inline README.md]: Then run the image, mount the directory as a Docker volume, and call the

Comment (dkravetz) [reply README.md]: "...and call the entrypoint". I added a line break to avoid long lines

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+set -o errexit
+set -o pipefail
+set -o nounset
+# set -o xtrace
+
+if [ $# -eq 0 ]; then
+    exec coordinator -f test_array.npy --host ${HOST} --port ${PORT}
+    else
+        exec coordinator "$@"
+    esle
```
Comment (PanicButtonPressed) [inline docker/entrypoint.sh]: esle?

Comment (dkravetz) [reply docker/entrypoint.sh]: typo on my part for another check I had before. Removing it, good catch!

**Pull request merged**